electrum/gui/qt/__init__.py

```diff
@@ -398,26 +403,65 @@ def start_new_window(
         return window
 
     def _start_wizard_to_select_or_create_wallet(self, path) -> Optional[Abstract_Wallet]:
-        wizard = InstallWizard(self.config, self.app, self.plugins, gui_object=self)
-        try:
-            path, storage = wizard.select_storage(path, self.daemon.get_wallet)
-            # storage is None if file does not exist
-            if storage is None:
-                wizard.path = path  # needed by trustedcoin plugin
-                wizard.run('new')
-                storage, db = wizard.create_storage(path)
-            else:
-                db = WalletDB(storage.read(), storage=storage, manual_upgrades=False)
-                wizard.run_upgrades(storage, db)
-        except (UserCancelled, GoBack):
-            return
-        except WalletAlreadyOpenInMemory as e:
-            return e.wallet
-        finally:
-            wizard.terminate()
-        # return if wallet creation is not complete
-        if storage is None or db.get_action():
+        wizard = QENewWalletWizard(self.config, self.app, self.plugins, self.daemon, path)
+        result = wizard.exec()
+        # TODO: use dialog.open() instead to avoid new event loop spawn?
+        self.logger.info(f'{result}')
+        if result == QENewWalletWizard.Rejected:
+            self.logger.info('ok bye bye')
             return
+
+        d = wizard.get_wizard_data()
+
+        if d['wallet_is_open']:
+            for window in self.windows:
+                if window.wallet.storage.path == d['wallet_name']:
+                    return window.wallet
+            raise Exception('found by wizard but not here?!')
+
+        if not d['wallet_exists']:
+            self.logger.info('about to create wallet')
+            wizard.create_storage()
+            if d['wallet_type'] == '2fa' and 'x3/' not in d:
+                return
+            wallet_file = wizard.path
+        else:
+            wallet_file = d['wallet_name']
+
+        storage = WalletStorage(wallet_file)
+        if storage.is_encrypted_with_user_pw() or storage.is_encrypted_with_hw_device():
+            storage.decrypt(d['password'])
+
+        db = WalletDB(storage.read(), storage=storage, manual_upgrades=True)
```

Comment:
Actually, `WalletDB` is called in daemon.py with `manual_upgrades=True`
This line is reached only after the user has clicked 'next' in the wizard first screen. 
Thus, `manual_upgrades` was correctly set to `False` in master, and should remain so.

Reply:
edit: I realize that run_wizard from installwizard.py is not used anymore, the code has moved

Reply:
manual_upgrades=True is just making sure no implicit upgrade is happening in the constructor. As I am here just using WalletDB to test for requires_split() and requires_upgrade() and eventually deferring the actual upgrade to wizard.run_upgrades() where I can show an in progress dialog.